lms/djangoapps/teams/csv.py

```diff
@@ -306,6 +314,39 @@ def validate_user_assignment_to_team_and_teamset(self, row):
                 return False
         return True
 
+    def validate_compatible_enrollment_modes(self, user, team_name, teamset_id):
+        """
+        Validates that only students enrolled in a masters track are on a single team. Disallows mixing of masters
+        with other enrollment modes on a single team.
+        """
+        if(teamset_id, team_name) not in self.user_enrollment_by_team:
+            self.user_enrollment_by_team[teamset_id, team_name] = set()
+        self.user_enrollment_by_team[teamset_id, team_name].add(self.user_to_actual_enrollment_mode[user.id])
+        if self.is_FERPA_bubble_breached(teamset_id, team_name):
+            error_message = \
+                'Team {} cannot have Master’s track users mixed with users in other tracks.'.format(team_name)
+            self.add_error_and_check_if_max_exceeded(error_message)
+            return False
+        return True
+
+    def is_FERPA_bubble_breached(self, teamset_id, team_name):
+        """
+        Ensures that FERPA bubble is not breached.
+        Checks that we are not trying to violate FERPA proctection by mixing masters
+        track students with other enrollment tracks.
+        """
+        protected_set = set()
+        for mode in ORGANIZATION_PROTECTED_MODES:
+            protected_set.add(mode)
+        if self.user_enrollment_by_team[teamset_id, team_name] - protected_set \
```

Comment:
sets have some useful descriptive methods that do basically the same things and i think would be a little easier to read,


```
if team_enrollment_modes.is_disjoint(protected_set):
   # all unprotected
elif team_enrollment_modes == protected_set:
  # all protected
else:
 # some mix


```

Reply:
actually, since the protected set could theoretically be more than just masters, 
```
elif team_enrllment_modes.is_subset(protected_set):
```